libafl/src/stages/sync.rs

```
@@ -75,7 +75,7 @@ impl<CB, E, EM, I, S, Z> Named for SyncFromDiskStage<CB, E, EM, I, S, Z> {
 
 impl<CB, E, EM, I, S, Z> Stage<E, EM, S, Z> for SyncFromDiskStage<CB, E, EM, I, S, Z>
 where
-    CB: FnMut(&mut Z, &mut S, &Path) -> Result<I, Error>,
+    CB: FnMut(&mut Z, &mut S, &Path) -> Result<Option<I>, Error>,
```

As discussed, I would say this should return the appropriate error